crates/vm/src/arch/execution_control.rs

```diff
@@ -12,38 +16,40 @@ where
     /// Host context
     type Ctx;
 
+    fn initialize_context(&self) -> Self::Ctx;
```

Comment:
i feel like we might need a different constructor for each `ctx` with different arguments. for e3, the context will have the trace buffers and would need the trace heights and widths as input during initialization
for your approach to work, we need to keep everything that is needed to initialize `ctx` in execution control. maybe that's fine but it's's not clear to me atm